### PR TITLE
docs: rework docs into its own directory

### LINKED ISSUE #25: docs: enhance documentation based on feedback
**Description**

The documentation is limited in features, this can be expanded upon.

### PR DESCRIPTION
**Pull request by wolfenrain**
## Description

Reworking the documentation to be more useful and structured.

Closes #25 

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by renancaraujo — DISMISSED** [review]

**Review by felangel — DISMISSED** [review]

File: .github/cspell.json [hunk 1]
```diff
@@ -21,6 +21,7 @@
   "words": [
     "Hitbox",
     "gameplay",
-    "collidable"
+    "collidable",
+    "Draggables"
```
Comment (scarletteliza) [inline .github/cspell.json:25]: TIL

File: README.md [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-Developed with 💙 by [Very Good Ventures][very_good_ventures_link] 🦄
-
 [![ci][ci_badge]][ci_link]
 [![coverage][coverage_badge]][ci_link]
 [![pub package][pub_badge]][pub_link]
 [![style: very good analysis][very_good_analysis_badge]][very_good_analysis_link]
 [![License: MIT][license_badge]][license_link]
 [![Powered by Flame][flame_badge_link]]([flame_link])
 
+Flame Behaviors applies separation of concerns to game logic in the form of Entities and Behaviors.
```
Comment (felangel) [inline README.md:10]:
```suggestion
Flame Behaviors applies separation of concerns to game logic in the form of entities and behaviors.
```

minor but was the capitalization intentional?

File: docs/collision-detection.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Collision Detection 💥
+
+Flame comes with a powerful built-in [collision detection system](https://docs.flame-engine.org/1.6.0/flame/collision_detection.html),
+but this API is not strongly typed. Components always get the colliding component as a
+`PositionComponent` and developers need to manually check what type of class it is.
+
+`flame_behaviors` is all about enforcing a strongly typed API. It provides a special behavior
+called `CollisionBehavior` that describes what type of entity it will target for collision. It
```
Comment (felangel) [inline docs/collision-detection.md:8]:
```suggestion
called `CollisionBehavior` that describes the type of entity being targeted for collision. It
```

File: docs/collision-detection.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+component that the parent entity is colliding with contains the target entity type specified in
+`CollisionBehavior`.
+
+By letting the `PropagatingCollisionBehavior` handle the collision detection we gain two benefits,
+the first and most important one is performance. By only registering collision callbacks on the
+entities themselves, the collision detection system does not have to go through any "collidable"
+behaviors, for which there could be many per entity. We only do that now if we confirm a collision
+has happened.
+
+The second benefit is that it allows for [separation of concerns][separation_of_concerns].
```
Comment (felangel) [inline docs/collision-detection.md:23]:
```suggestion
The second benefit is that it allows for [separation of concerns][separation_of_concerns].
```

Aren't we missing the link for `separation_of_concerns` here?

File: docs/conventions/coding-conventions.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
+# Coding Conventions
+
+> **Note**: The following coding conventions are simply recommendations and are completely
+optional. Feel free to use whatever coding conventions you prefer.
+
+## Entities
+
+Entities should not contain any behavioral logic, instead they should be composed of behaviors. This
+allows for more flexible and reusable code. Entities should not do any direct rendering, instead they
+should add child components to themselves to do the visualization of the entity.
```
Comment (felangel) [inline docs/conventions/coding-conventions.md:10]:
```suggestion
should add child components to handle the visualization of the entity.
```

File: docs/conventions/coding-conventions.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+## Behaviors
+
+Behaviors should only contain code related to the behavioral logic it describes. Behaviors should 
+never do any direct rendering.
+
+A behavior is allowed to have its own components for adding extra functionality related to the 
+behavior. For example, a behavior that makes an entity jump could have a `TimerComponent` to ensure
+that the entity can only jump once every 0.5 seconds. And that behavior can also use a 
+`KeyboardHandler` mixin to listen for the jump key to trigger the jump. But any logic that is not
```
Comment (scarletteliza) [inline docs/conventions/coding-conventions.md:58]:
```suggestion
`KeyboardHandler` mixin to listen for the jump key to trigger the jump. Any logic that is not
```

File: docs/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+multiple `Behavior`s, which in turn define how the entity behaves.
+
+```dart
+// Define a custom entity by extending `Entity`.
+class MyEntity extends Entity {
+  MyEntity() : super(behaviors: [MyBehavior()]);
+}
+```
+
+### Types of entities
```
Comment (scarletteliza) [inline docs/getting-started.md:31]:
```suggestion
### Types of Entities
```

File: docs/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  ...
+}
+
+// Can only be added to MyEntity and subclasses of it.
+class MySpecificBehavior extends Behavior<MyEntity> {
+  ...
+}
+```
+
+### Behavior composition
```
Comment (scarletteliza) [inline docs/getting-started.md:82]:
```suggestion
### Behavior Composition
```

File: docs/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+  void _onTick() {
+    // Do something every 5 seconds.
+  }
+}
+```
+
+**Note**: A `Behavior` is a non-visual component that describes how a visual component (Entity)
+behaves. To ensure this rule, a behavior can't have its own `Behavior`s.
+
+## Whats next
```
Comment (scarletteliza) [inline docs/getting-started.md:103]:
```suggestion
## What's Next
```

File: docs/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+The following sections will show you how to use Flame Behaviors for common game development tasks:
+- [Handling game input][input_behaviors_link]
+- [Handling collisions][collision_behaviors_link
```
Comment (scarletteliza) [inline docs/getting-started.md:107]:
```suggestion
- [Handling Game Input][input_behaviors_link]
- [Handling Collisions][collision_behaviors_link
```

File: docs/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+  }
+
+  void _onTick() {
+    // Do something every 5 seconds.
+  }
+}
+```
+
+**Note**: A `Behavior` is a non-visual component that describes how a visual component (Entity)
+behaves. To ensure this rule, a behavior can't have its own `Behavior`s.
```
Comment (felangel) [inline docs/getting-started.md:101]:
```suggestion
behaves therefore, a behavior can't have its own `Behavior`s.
```

File: packages/flame_behaviors/pubspec.yaml [hunk 12]
```diff
@@ -4,7 +4,7 @@ version: 0.1.1
 homepage: https://github.com/VeryGoodOpenSource/flame_behaviors
 repository: https://github.com/VeryGoodOpenSource/flame_behaviors
 issue_tracker: https://github.com/VeryGoodOpenSource/flame_behaviors/issues
-# documentation: https://github.com/VeryGoodOpenSource/flame_behaviors/tree/main/docs
+documentation: https://github.com/VeryGoodOpenSource/flame_behaviors/tree/main/docs
```
Comment (felangel) [inline packages/flame_behaviors/pubspec.yaml:7]: 💯

File: docs/getting-started.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+
+The following sections will show you how to use Flame Behaviors for common game development tasks:
+- [Handling game input][input_behaviors_link]
+- [Handling collisions][collision_behaviors_link
+
+Flame Behaviors also provides some conventions on how to name and organize your code:
+- [Naming conventions][naming_conventions_link]
+- [Code conventions][code_conventions_link]
+
+If you want to learn more about how to use Flame Behaviors, you can check out the [article][article_link].
```
Comment (scarletteliza) [inline docs/getting-started.md:113]:
```suggestion
To learn more about how to use Flame Behaviors, check out our [article][article_link].
```

File: docs/input-behaviors.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
+# Input behaviors ⌨
```
Comment (scarletteliza) [inline docs/input-behaviors.md:1]:
```suggestion
# Input Behaviors ⌨
```

File: docs/input-behaviors.md [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+The `flame_behaviors` package also provides input behaviors. These behaviors are a layer over the
+existing Flame input mixins for components. These behaviors will trigger when their parent entity
+is being interacted with by the user. So these events are always relative to the parent entity.
```
Comment (scarletteliza) [inline docs/input-behaviors.md:5]:
```suggestion
is being interacted with by the user. These events are always relative to the parent entity.
```

File: docs/input-behaviors.md [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+## DraggableBehavior
+
+The `DraggableBehavior` brings the [drag events][flame_drag_docs] from Flame into a behavioral
+format.
```
Comment (scarletteliza) [inline docs/input-behaviors.md:10]:
```suggestion
The `DraggableBehavior` brings the [drag events][flame_drag_docs] from Flame into a behavioral
format.
```

but what does it mean?

File: README.md (hunk 2 above)
Comment (wolfenrain) [reply README.md:10]: Well it was inconsistent compared to the rest of the docs, so I decided to do it like this

File: docs/collision-detection.md (hunk 4 above)
Comment (wolfenrain) [reply docs/collision-detection.md:23]: 👀  maybe? I'll check up what it was originally

File: docs/input-behaviors.md (hunk 16 above)
Comment (wolfenrain) [reply docs/input-behaviors.md:10]: That the `DraggableBehavior` brings the drag events from Flame into a behavioral format!

Will try to add a better context/explanation to it!

File: docs/collision-detection.md (hunk 4 above)
Comment (wolfenrain) [reply docs/collision-detection.md:23]: Will be added in next commit!

**Review by felangel — APPROVED** [review]

**Review by scarletteliza — APPROVED** [review]

**Pull request merged**